components/pysd_gen/gen_pysd.c

```diff
@@ -0,0 +1,89 @@
+    }
+
+    return string_size;
+}
+
+/**
+ * @brief Fill buffer with struct variables names
+ * 
+ * @param buffer pointer to buffer
+ * @param size buffer size
+ * @param rocket_data_t struct with data
+ * @param new_line_ending end string with new line character
+ * @return size_t size of wrote data, return 0 in case of failure
+ */
+size_t pysd_create_header(char *buffer, size_t size, bool new_line_ending) {
+    size_t header_size;
+    header_size = pysd_get_header_size(new_line_ending);
+
+    if (header_size > size) {
+        return false;
+    }
+
+    if (new_line_ending == true) {
+        snprintf(buffer, size, "mcb state;mcb uptime;mcb flight_time;mcb battery_voltage;mcb sensor_placeholder;vent_valve waken_up;vent_valve valve_state;vent_valve thermistor1;vent_valve thermistor2;vent_valve tank_pressure;vent_valve battery_voltage;main_valve waken_up;main_valve valve_state;main_valve thermocouple1;main_valve thermocouple2;main_valve battery_voltage;recovery isArmed;recovery isTeleActive;recovery easyMiniFirstStage;recovery easyMiniSecondStage;recovery telemetrumFirstStage;recovery telemetrumSecondStage;recovery firstStageDone;recovery secondStageDone;recovery firstStageContinouity;recovery secondStageContinouity;recovery separationSwitch1;recovery separationSwitch2;recovery pressure1;recovery pressure2;\n");
+    } else {
+        snprintf(buffer, size, "mcb state;mcb uptime;mcb flight_time;mcb battery_voltage;mcb sensor_placeholder;vent_valve waken_up;vent_valve valve_state;vent_valve thermistor1;vent_valve thermistor2;vent_valve tank_pressure;vent_valve battery_voltage;main_valve waken_up;main_valve valve_state;main_valve thermocouple1;main_valve thermocouple2;main_valve battery_voltage;recovery isArmed;recovery isTeleActive;recovery easyMiniFirstStage;recovery easyMiniSecondStage;recovery telemetrumFirstStage;recovery telemetrumSecondStage;recovery firstStageDone;recovery secondStageDone;recovery firstStageContinouity;recovery secondStageContinouity;recovery separationSwitch1;recovery separationSwitch2;recovery pressure1;recovery pressure2;");
+    }
+
+    return header_size;
+}
+
+/**
+ * @brief Fill buffer with struct data
+ *
+ * @param buffer pointer to buffer
+ * @param size buffer size
+ * @param ...pysd_main... struct with data
+ * @param new_line_ending end string with new line character
+ * @return size_t size of wrote data, return 0 in case of failure
+ */
+size_t pysd_create_sd_frame(char *buffer, size_t size, rocket_data_t pysd_main, bool new_line_ending) {
+    size_t frame_size;
+    frame_size = pysd_get_sd_frame_size(pysd_main, new_line_ending);
```

Comment:
pysd_get_frame_size robi dokladnie to samo co pysd_create_sd_frame z jedna roznica ze zwraca dlugosc ramki, mozesz to miec w jednej funkcji i bedzie zajmowac polowe czasu